### PR TITLE
Intake System 

### LINKED ISSUE #19: Implement basic intake functions and commands
Implement basic intake subsystem (1 spark max motor controller) and the following basic commands to:

* CubeIntake
* CubeDrop
* ConeIntake
* ConeDrop

### PR DESCRIPTION
**Pull request by jagernet-ops**
Configured the intake subsystem into what I believe to be a workable state.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -15,5 +15,16 @@
...
   public static class OperatorConstants {
     public static final int kDriverControllerPort = 0;
+  
   }
+  /*
+  All changes in the constants file are for testing and will be integrated into the larger constants file
+  further down the line by the programming team :)
+  */
+  
+  public static final int INTAKEMOTOR = 9;
```
Comment (jherrmann06) [inline src/main/java/frc/robot/Constants.java]: Use CAPS_UNDERSCORE naming conventions for constants

File: src/main/java/frc/robot/commands/ConeUtil.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.Intake;
+
+public class ConeUtil extends CommandBase {
```
Comment (jherrmann06) [inline src/main/java/frc/robot/commands/ConeUtil.java]: More descriptive name like `IntakeConeOutputCube` or something

File: src/main/java/frc/robot/commands/CubeUtil.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+import frc.robot.Constants;
+import frc.robot.subsystems.Intake;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class CubeUtil extends CommandBase {
```
Comment (jherrmann06) [inline src/main/java/frc/robot/commands/CubeUtil.java]: Naming again, see above

**Review by jherrmann06 — COMMENTED** [review]
Good work, just a couple of changes. Also need to fix merge conflicts. Do that in VSCode not on Github.

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -3,14 +3,26 @@
...
 
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ConditionalCommand;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
```
Comment (jdlanning) [inline src/main/java/frc/robot/RobotContainer.java]: We can do an Alt+Shift+O to clean up unused imports

File: src/main/java/frc/robot/commands/CubeUtil.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    intake.conePutCubePick(Constants.INTAKESPEED);
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {}
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/CubeUtil.java]: Need to set the speed to 0 in end()

File: src/main/java/frc/robot/commands/CubeUtil.java (hunk 3 above)
Comment (jdlanning) [reply src/main/java/frc/robot/commands/CubeUtil.java]: Agree with Jake, lets create 4 different commands (IntakeCube, IntakeCone, DropCube, DropCone)

File: src/main/java/frc/robot/commands/ConeUtil.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    intake.conePickCubePut(Constants.INTAKESPEED);
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {}
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/ConeUtil.java]: Needs to set motor to 0 here

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  private CANSparkMax intakeMotor = new CANSparkMax( Constants.INTAKEMOTOR, MotorType.kBrushless);
+  public Intake() {
+
+  }
+
+  public void conePickCubePut(double speed){
+    intakeMotor.set(speed * -1);
+  }
+
+  public void conePutCubePick(double speed){
```
Comment (jdlanning) [inline src/main/java/frc/robot/subsystems/Intake.java]: Lets implement 4 different methods here would be nice intakeCone, intakeCube, dropCone, dropCube

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (jdlanning) [reply src/main/java/frc/robot/Constants.java]: After you merge group this with the other constants for motors

File: src/main/java/frc/robot/Robot.java [hunk 8]
```diff
@@ -27,7 +25,6 @@ public class Robot extends TimedRobot {
   public void robotInit() {
     // Instantiate our RobotContainer.  This will perform all our button bindings, and put our
     // autonomous chooser on the dashboard.
-    m_robotContainer = new RobotContainer();
```
Comment (jdlanning) [inline src/main/java/frc/robot/Robot.java]: Looks like we lost robot container somehow. we should revert back to original Robot.java

File: src/main/java/frc/robot/commands/DropCube.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    intake.cubeDrop(Constants.INTAKE_SPEED);
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {}
```
Comment (jdlanning) [inline src/main/java/frc/robot/commands/DropCube.java]: Need to set all of these commands to turn off the motor when ending

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    intakeMotor.set(speed * 1);
+  }
+
+  public void cubeDrop(double speed){
+    intakeMotor.set(speed * -1);
+  }
+
+  public void cubePickup(double speed){
+    intakeMotor.set(speed * 1);
+  }
```
Comment (jdlanning) [inline src/main/java/frc/robot/subsystems/Intake.java:37]: Need to add a stop method to this subsystem to be called from the end of each command

**Review by jdlanning — APPROVED** [review]

**Pull request merged**